pkg/utils/connections/connection.go

```diff
@@ -239,12 +239,22 @@ func saveConnectionsConfigFile(ConnectionConfig *ConnectionConfig) *ConError {
 
 // getConnectionConfigDir : get directory path to the connections file
 func getConnectionConfigDir() string {
-	const GOOS string = runtime.GOOS
+	val, ok := os.LookupEnv("CHE_API_EXTERNAL")
 	homeDir := ""
-	if GOOS == "windows" {
-		homeDir = os.Getenv("USERPROFILE")
+	if ok && (val != "") {
+		val, ok := os.LookupEnv("CHE_PROJECTS_ROOT")
+		if ok && (val != "") {
+			homeDir = val
+		} else {
+			panic("CHE_PROJECTS_ROOT not set")
```

Comment:
We don't usually want to use panics but this is necessary in this case. Please can this have a comment to explain and prevent it being removed by accident in future?

Reply:
Short explanation that an unset env variable suggests a wider issue with Che install of codewind